tcp.go

```diff
@@ -25,8 +27,27 @@ type TCPTun interface {
 
 var truncated_error error = errors.New("truncated packet")
 
-func NewTCPTunnel(tun TCPTun, d DataConnection, dest types.NodeAddress, amt int64) *TCP {
+func SetDevAddr(dev string, addr string) error {
+	_, err := exec.Command("ip", "addr", "add", addr, "dev", dev).CombinedOutput()
+	return err
+}
+
+func NewTCPTunnel(tun TCPTun, d DataConnection, dest types.NodeAddress, amt int64, name string) *TCP {
 	t := &TCP{d, tun, dest, amt, make(chan bool), make(chan error, 1)}
+	ep := types.Packet{dest, amt, "hello"}
+	go func() {
+		err := t.data.SendPacket(ep)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	p := <-t.data.Packets()
+
+	err := SetDevAddr(name, p.Data)
```

Comment:
Can we add a little bit of validation / a header? At a bare minimum a version int at the beginning will make all our lives easier.